### PR TITLE
Fix: Fixed crash when opening folders with invalid git configs

### PR DESCRIPTION
**Pull request by yaira2**
**Resolved / Related Issues**

To prevent extra work, all changes to the Files codebase must link to an approved issue marked as `Ready to build`. Please insert the issue number following the hashtag with the issue number that this Pull Request resolves.
- Closes https://files-org.sentry.io/issues/5549705044/events/95f583157753427d9838cdb159b84d32/

**Steps used to test these changes**

Stability is a top priority for Files and all changes are required to go through testing before being merged into the repo. Please include a list of steps that you used to test this PR.

1. Opened Files ...
2. ...

### CODE REVIEW

**Review by hishitetsu — COMMENTED** [review]
* You can use `SafetyExtensions.IgnoreExceptions` as it returns false when an exception is thrown.
* Is there no need to consider other places that use `Repository.IsValid`?

**Comment by yaira2** [comment]
> You can use SafetyExtensions.IgnoreExceptions as it returns false when an exception is thrown.

We have to return a string so I think try catch makes sense in this case.

File: src/Files.App/Utils/Git/GitHelpers.cs [hunk 1]
```diff
@@ -646,11 +709,17 @@ public static bool IsRepositoryEx(string path, out string repoRootPath)
...
-			if (Repository.IsValid(repositoryRootPath))
+			// Workaround for https://github.com/libgit2/libgit2sharp/issues/1851
+
+			try
 			{
-				repoRootPath = repositoryRootPath;
-				return true;
+				if (Repository.IsValid(repositoryRootPath))
+				{
+					repoRootPath = repositoryRootPath;
```
Comment (yaira2) [inline src/Files.App/Utils/Git/GitHelpers.cs:718]: I would use `SafetyExtensions.IgnoreExceptions` here but it doesn't allow access to `repoRootPath`.

File: src/Files.App/Utils/Git/GitHelpers.cs [hunk 2]
```diff
@@ -99,8 +97,19 @@ private set
...
+			if (string.IsNullOrWhiteSpace(path))
+				return string.Empty;
+
+			// Workaround for https://github.com/libgit2/libgit2sharp/issues/1851
+			try
+			{
+				if (!Repository.IsValid(path))
+					return string.Empty;
+			}
+			catch (Exception)
```
Comment (0x5bfa) [inline src/Files.App/Utils/Git/GitHelpers.cs:109]: Should catch specify Exception class derivatives?

**Comment by hishitetsu** [comment]
> > You can use SafetyExtensions.IgnoreExceptions as it returns false when an exception is thrown.
> 
> We have to return a string so I think try catch makes sense in this case.

I was thnking the following code could be included in the if condition.
`SafetyExtensions.IgnoreExceptions(() => !Repository.IsValid(path))`

**Comment by yaira2** [comment]
I didn't realize that works, it's certainly cleaner that way.

File: src/Files.App/Utils/Git/GitHelpers.cs [hunk 3]
```diff
@@ -78,7 +76,7 @@ private set
 
 			try
 			{
-				if (Repository.IsValid(path))
+				if (SafetyExtensions.IgnoreExceptions(() => Repository.IsValid(path)))
```
Comment (hishitetsu) [inline src/Files.App/Utils/Git/GitHelpers.cs:79]: It could be left as is, but it would be even better to cut it out into a separate method and call it.

Comment (yaira2) [reply src/Files.App/Utils/Git/GitHelpers.cs:79]: What's the benefit of doing it like that?

Comment (hishitetsu) [reply src/Files.App/Utils/Git/GitHelpers.cs:79]: Don't have to surround each call to `isValid` with `SafetyExtensions.IgnoreExceptions`.

Comment (yaira2) [reply src/Files.App/Utils/Git/GitHelpers.cs]: Done 👍

**Review by hishitetsu — APPROVED** [review]
LGTM👍

**Pull request merged**